File: app/src/main/java/com/example/heriprastio/recyclerviewinfragment/adapter/MyAdapter.java
package com.example.heriprastio.recyclerviewinfragment.adapter;

import android.support.annotation.NonNull;
import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import com.example.heriprastio.recyclerviewinfragment.R;

public class MyAdapter extends RecyclerView.Adapter<MyAdapter.MyViewHolder> {

    private String[] mDataSet;

    public MyAdapter(String[] strings) {
        this.mDataSet = strings;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View v = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.card_item, viewGroup, false);
        MyViewHolder viewHolder = new MyViewHolder(v);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder myViewHolder, int i) {

        myViewHolder.mTextView.setText(mDataSet[i]);
    }

    @Override
    public int getItemCount() {
        return mDataSet.length;
    }

    class MyViewHolder extends RecyclerView.ViewHolder{
        public CardView mCadView;
        public TextView mTextView;

        public MyViewHolder(@NonNull View itemView) {
            super(itemView);
            mCadView = (CardView) itemView.findViewById(R.id.card_view);
            mTextView = (TextView) itemView.findViewById(R.id.tv_blah);
        }
    }
}

File: app/src/main/java/com/example/heriprastio/recyclerviewinfragment/adapter/MyAdapter.kt
package com.example.heriprastio.recyclerviewinfragment.adapter

import android.support.v7.widget.CardView
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.example.heriprastio.recyclerviewinfragment.R

class MyAdapter(private val mDataSet: Array<String>) : RecyclerView.Adapter<MyAdapter.MyViewHolder>() {

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int) =
    MyViewHolder(LayoutInflater.from(viewGroup.context).inflate(R.layout.card_item, viewGroup, false))


    override fun onBindViewHolder(holder: MyViewHolder, i: Int) {

        holder.mTextView.text = mDataSet[i]
    }

    override fun getItemCount(): Int {
        return mDataSet.size
    }

    inner class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var mCadView: CardView
        var mTextView: TextView

        init {
            mCadView = itemView.findViewById<View>(R.id.card_view) as CardView
            mTextView = itemView.findViewById<View>(R.id.tv_blah) as TextView
        }
    }
}
